src/init.cpp

```diff
@@ -1488,6 +1488,13 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         uiInterface.NotifyBlockTip.disconnect(BlockNotifyGenesisWait);
     }
 
+#ifdef ENABLE_WALLET
```

Comment:
It could be combined with the guard a few lines below, but this won't solve the larger problem.

I think for a bugfix this is fine.